apps/patient/src/components/CouponDetails.tsx

```diff
@@ -0,0 +1,58 @@
+import { Box, HStack, Icon, Text, VStack } from '@chakra-ui/react';
+import { FC, useState } from 'react';
+import { FiInfo } from 'react-icons/fi';
+import { CouponModal } from '../components';
+import { text as t } from '../utils/text';
+import { useOrderContext } from '../views/Main';
+
+export const CouponDetails: FC = () => {
+  const [couponModalOpen, setCouponModalOpen] = useState<boolean>(false);
+  const { order } = useOrderContext();
+
+  const { price, bin, pcn, group, memberId } = order.discountCards[0];
```

Comment:
This is a place where typescript falls over. `order.discountCards[0]` can be undefined and this would fall over. So we should first test if length > 0

Reply:
oh thought i had a check for that